src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -90,7 +91,26 @@ public RMProfileClient(AzureRMProfile profile)
                     try
                     {
                         token = AcquireAccessToken(account, environment, tenant, password, ShowDialog.Auto);
-                        account.SetOrAppendProperty(AzureAccount.Property.Tenants, tenant);
+
+                        if (accountId == null)
+                        {
+                            accountId = account.Id;
+                            account.SetOrAppendProperty(AzureAccount.Property.Tenants, tenant);
+                        }
+                        else if (accountId == account.Id)
```

Comment:
We probably want to control the strong comparison, I would use string.Equals, with StringComparison.OrdinalIgnoreCase